### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by samturrell**
Apologies for the large changelog again!

This PR includes:

##### ES6 module structure
I've moved helper functions and some static data out of the main source file and into their own respective `utils/*.js` files. I believe this kind of approach will help in the long run as more features are added, the main scope isn't going to be completely bloated. I'd ideally like to move specific features out into their own files so their logic is contained and easy to maintain, but I haven't put much thought into how to handle this just yet.

##### Code cleanup
I've removed the main lengthy block of code that sets the users provided options, and replaced it with the `extend()` helper. Let me know if you see any problems with this. HoundCI hates this block, and so do I 😄 

In Summary: Code is a headache, let's clean it up and modularise it.

### CODE REVIEW

**Comment by limonte** [comment]
Wow! I'm very grateful to you @samturrell for this PR! This is the best refactoring PR on this project.

**Comment by limonte** [comment]
[v3.2.0](https://github.com/limonte/sweetalert2/releases/tag/v3.2.0)

**Comment by limonte** [comment]
Hi @samturrell ! 
I fixed most JS coding style errors in https://github.com/limonte/sweetalert2/commit/bc24ada5e183511731a627a6148c9e7e2a7f8481

Also, I added `gulp-eslint` to automate JS validation in https://github.com/limonte/sweetalert2/commit/e0b89e450b2be1bd71c47be7c57213231338513b 

But I'm still not enough familiar with modules and not sure how to fix these:

**Comment by samturrell** [comment]
Hey @limonte, fixed in 0dc4e63

**Comment by limonte** [comment]
@samturrell nicely done!
